### PR TITLE
APR-1656 API endpoints for PageReviewStatuses

### PR DESCRIPTION
**Pull request by MakCom**

### CODE REVIEW

File: src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpAssessorController.cs [hunk 1]
```diff
@@ -149,12 +149,99 @@ private static string GetSequenceTitle(int sequenceId)
...
+                                                                                request.SectionNumber,
+                                                                                request.PageId,
+                                                                                request.AssessorType,
+                                                                                request.UserId,
+                                                                                request.Status,
+                                                                                request.Comment));
+        }
+
+        [HttpPost("Assessor/GetPageReviewOutcome")]
+        public async Task<PageReviewOutcome> GetPageReviewOutcome([FromBody] GetPageReviewOutcomeRequest request)
```
Comment (martin-hornsby) [inline src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpAssessorController.cs:167]: This method says 'Get' yet you're using a HttpPost to retrieve information

File: src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpAssessorController.cs [hunk 2]
```diff
@@ -149,12 +149,99 @@ private static string GetSequenceTitle(int sequenceId)
...
+                                                                                request.SequenceNumber,
+                                                                                request.SectionNumber,
+                                                                                request.AssessorType,
+                                                                                request.UserId));
+
+            return assessorReviewOutcomes;
+        }
+
+        [HttpPost("Assessor/GetAllAssessorReviewOutcomes")]
+        public async Task<List<PageReviewOutcome>> GetAllAssessorReviewOutcomes([FromBody] GetAllAssessorReviewOutcomesRequest request)
```
Comment (martin-hornsby) [inline src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpAssessorController.cs:192]: This method says 'Get' yet you're using a HttpPost to retrieve information

File: src/SFA.DAS.ApplyService.Data/ApplyRepository.cs [hunk 3]
```diff
@@ -1226,5 +1226,238 @@ public async Task<bool> UpdateApplyData(Guid applicationId, ApplyData applyData,
...
+			                                                                  ,[Assessor2ReviewComment] AS Comment
+		                                                                  FROM [dbo].[AssessorPageReviewOutcome]
+		                                                                  WHERE [ApplicationId] = @applicationId AND
+				                                                                [Assessor2UserId] = @userId                      
+	                                                                END",
+                    new { applicationId, assessorType, userId });
+
+                return pageReviewOutcomeResults.ToList();
+            }
+        }
```
Comment (David-B-Read) [inline src/SFA.DAS.ApplyService.Data/ApplyRepository.cs:1461]: ApplyRepository is getting massive now. I'll raise a tech debt ticket to look at splitting it up

File: src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpAssessorController.cs [hunk 4]
```diff
@@ -149,12 +149,99 @@ private static string GetSequenceTitle(int sequenceId)
                     return null;
             }
         }
+
+        [HttpPost("Assessor/SubmitPageOutcome")]
+        public async Task SubmitAssessorPageOutcome([FromBody] SubmitAssessorPageOutcomeRequest request)
+        {
+            await _mediator.Send(new SubmitAssessorPageOutcomeHandlerRequest(request.ApplicationId,
```
Comment (David-B-Read) [inline src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpAssessorController.cs:156]: What's the value in creating one request type from another? The handler could use SubmitAssessorPageOutcomeRequest?

Comment (MakCom) [reply src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpAssessorController.cs:156]: I couldn’t see it either, but then I saw that this is the approach taken in AssignApplication() method.
It uses AssignAssessorApplicationRequest and AssignAssessorRequest. (applicationId could be included in AssignAssessorApplicationRequest and they will become the same)
Plus, …HandlerRequests inherit from IRequest<T>. The other ones don’t. 
And I have decided to follow the same pattern.

File: src/SFA.DAS.ApplyService.Data/ApplyRepository.cs (hunk 3 above)
Comment (MakCom) [reply src/SFA.DAS.ApplyService.Data/ApplyRepository.cs:1461]: Yes, It's about time to do that

File: src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpAssessorController.cs (hunk 4 above)
Comment (MakCom) [reply src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpAssessorController.cs]: The two request objects are merged into one.

File: src/SFA.DAS.ApplyService.Data/ApplyRepository.cs (hunk 3 above)
Comment (MichaelYoung1981) [reply src/SFA.DAS.ApplyService.Data/ApplyRepository.cs:1461]: Why don't we do it as part of this ticket then?  I've already created an AssessorRepository so maybe these methods could go on there?

File: src/SFA.DAS.ApplyService.Application/Apply/Assessor/GetPageReviewOutcomeRequest.cs [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        {
+            ApplicationId = applicationId;
+            SequenceNumber = sequenceNumber;
+            SectionNumber = sectionNumber;
+            PageId = pageId;
+            AssessorType = assessorType;
+            UserId = userId;
+        }
+
+        public Guid ApplicationId { get; set; }
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.ApplyService.Application/Apply/Assessor/GetPageReviewOutcomeRequest.cs:31]: Can the properties be made Get only if they're being set in the constructor?  Same applies to the morder Request classes.

File: src/SFA.DAS.ApplyService.Data/ApplyRepository.cs (hunk 3 above)
Comment (MakCom) [reply src/SFA.DAS.ApplyService.Data/ApplyRepository.cs:1461]: Methods moved to AssessorRepository => PR 549

File: src/SFA.DAS.ApplyService.Application/Apply/Assessor/GetPageReviewOutcomeRequest.cs (hunk 5 above)
Comment (MakCom) [reply src/SFA.DAS.ApplyService.Application/Apply/Assessor/GetPageReviewOutcomeRequest.cs:31]: Setters removed => PR 549

**Pull request merged**